src/speak-tts.js

```diff
@@ -166,8 +188,8 @@ class SpeakTTS {
         if(this.pitch) utterance.pitch = this.pitch //0 to 2
         utterance.text = sentence
 
-        // Attach event listeners
-        Object.keys(listeners).forEach(listener => {
+        // Attach event listeners, and make sure onerror and onend are included
```

Comment:
How is that necessary  ?

Reply:
onerror & onend are used by speaktts to resolve/reject the speak promise, and this should happen even if the user did not pass customized versions of those liseteners into the init function.

Reply:
if no customized listener is passed, this is going to attach undefined variables ?

Reply:
No, newListener is a function that calls the user-provided listener (in this scenerio that is NULL and it won't be called due to "fn && fn(data)" ) but it also checks if the listener is for "onend" or "onerror" in order to resolve/reject the speak promise.

Prior to this change, if no custom listener for onend and onerror is passed, then the speak promise will not resolve or reject.

Reply:
I see, thanks for the explanation. I think it could be resolved by giving a noop function as a defaut value for onEnd et onError